geom/tests/Test_FieldBLAS.pf

```diff
@@ -0,0 +1,926 @@
+
+      YR4 = mk_field(y4array, regDecomp=REG_DECOMP_DEFAULT, minIndex=MIN_INDEX_DEFAULT, maxIndex=MAX_INDEX_DEFAULT, &
+         indexflag=INDEX_FLAG_DEFAULT, name = 'YR4', rc = status)
+
+      XR8 = mk_field(R8_ARRAY_DEFAULT, regDecomp=REG_DECOMP_DEFAULT, minIndex=MIN_INDEX_DEFAULT, maxIndex=MAX_INDEX_DEFAULT, &
+         indexflag=INDEX_FLAG_DEFAULT, name = 'XR8', rc = status)
+
+      YR8 = mk_field(y8array, regDecomp=REG_DECOMP_DEFAULT, minIndex=MIN_INDEX_DEFAULT, maxIndex=MAX_INDEX_DEFAULT, &
+         indexflag=INDEX_FLAG_DEFAULT, name = 'YR8', rc = status)
+
+   end subroutine set_up_data
+
+   @Test
+   ! Basic test of FieldCOPY subroutine (REAL32)
+   subroutine test_FieldCOPY_R4()
+      character(len=*), parameter :: TEST_NAME = 'test_FieldCOPY_R4'
+      type(ESMF_Field) :: x
+      type(ESMF_Field) :: y
+      real(kind=ESMF_KIND_R4), dimension(:,:), pointer :: x_ptr
+      real(kind=ESMF_KIND_R4), dimension(:,:), pointer :: y_ptr
+      integer :: status, rc
+      
+      RUNNING(TEST_NAME)
+
+      x = XR4
+      y = YR4
+
+      call FieldCOPY(x, y, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Copy failed', status))
+      
+      call ESMF_FieldGet(x, farrayPtr = x_ptr, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Failed to get source data pointer', status))
+
+      call ESMF_FieldGet(y, farrayPtr = y_ptr, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Failed to get destination data pointer', status))
+
+      @assertEqual(x_ptr, y_ptr)
+
+      PASSED(TEST_NAME)
+
+   end subroutine test_FieldCOPY_R4
+
+   @Test
+   ! Basic test of FieldCOPY subroutine (REAL64)
+   subroutine test_FieldCOPY_R8()
+      character(len=*), parameter :: TEST_NAME = 'test_FieldCOPY_R8'
+      type(ESMF_Field) :: x
+      type(ESMF_Field) :: y
+      real(kind=ESMF_KIND_R8), dimension(:,:), pointer :: x_ptr
+      real(kind=ESMF_KIND_R8), dimension(:,:), pointer :: y_ptr
+      integer :: status, rc
+      
+      RUNNING(TEST_NAME)
+
+      x = XR8 
+      y = YR8
+
+      call FieldCOPY(x, y, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Copy failed', status))
+      
+      call ESMF_FieldGet(x, farrayPtr = x_ptr, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Failed to get source data pointer', status))
+
+      call ESMF_FieldGet(y, farrayPtr = y_ptr, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Failed to get destination data pointer', status))
+
+      @assertEqual(x_ptr, y_ptr)
+
+      PASSED(TEST_NAME)
+
+   end subroutine test_FieldCOPY_R8
+
+   @Test
+   ! Basic test of FieldCOPY subroutine (REAL32 -> REAL64)
+   subroutine test_FieldCOPY_R4R8()
+      character(len=*), parameter :: TEST_NAME = 'test_FieldCOPY_R4R8'
+      type(ESMF_Field) :: x
+      type(ESMF_Field) :: y
+      real(kind=ESMF_KIND_R4), dimension(:,:), pointer :: x_ptr
+      real(kind=ESMF_KIND_R8), dimension(:,:), pointer :: y_ptr
+      integer :: status, rc
+      
+      RUNNING(TEST_NAME)
```

Comment:
Note that pFUnit knows the name of its tests, so you don't need to define `TEST_NAME` in each test.

Reply:
Removed.